nerfstudio/data/dataparsers/nerfstudio_dataparser.py

```diff
@@ -305,28 +307,98 @@ def _generate_dataparser_outputs(self, split="train"):
         assert self.downscale_factor is not None
         cameras.rescale_output_resolution(scaling_factor=1.0 / self.downscale_factor)
 
-        if "applied_transform" in meta:
-            applied_transform = torch.tensor(meta["applied_transform"], dtype=transform_matrix.dtype)
-            transform_matrix = transform_matrix @ torch.cat(
-                [applied_transform, torch.tensor([[0, 0, 0, 1]], dtype=transform_matrix.dtype)], 0
-            )
+        # The naming is somewhat confusing, but:
+        # - transform_matrix contains the transformation to dataparser output coordinates from saved coordinates.
+        # - dataparser_transform_matrix contains the transformation to dataparser output coordinates from original data coordinates.
+        # - applied_transform contains the transformation to saved coordinates from original data coordinates.
+        if "applied_transform" not in meta:
+            # For converting from colmap, this was the effective value of applied_transform that was being
+            # used before we added the applied_transform field to the output dataformat.
+            meta["applied_transform"] = [[0, 1, 0, 0], [1, 0, 0, 0], [0, 0, -1, 0]]
```

Comment:
@brentyi For my understanding, is this to backward support old converted nerfstudio format from colmap?

Reply:
yeah, that's right!

I ended up tweaking this logic slightly to prevent it from breaking converted datasets that didn't come from colmap, although the core logic is still the same:
https://github.com/nerfstudio-project/nerfstudio/commit/f55bd04ca6f16a85d7f12a479b94850228e3c7d5